processor/file.go

```diff
@@ -184,6 +177,21 @@ DIRENTS:
 	}
 }
 
+func (dw *DirectoryWalker) Readdir(path string) ([]os.FileInfo, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return []os.FileInfo{}, fmt.Errorf("failed to open %s: %v", path, err)
+	}
+	defer file.Close()
+
+	dirents, err := file.Readdir(-1)
```

Comment:
Whats the purpose of -1? Can we just put a comment on what this actually does.

Reply:
From https://golang.org/pkg/os/#File.Readdir :

> If n <= 0, Readdir returns all the FileInfo from the directory in a single slice. In this case, if Readdir succeeds (reads all the way to the end of the directory), it returns the slice and a nil error. If it encounters an error before the end of the directory, Readdir returns the FileInfo read until that point and a non-nil error.

Reply:
Ah cool. I was worried it was some magic number hack. Cheers.